src/__tests__/autocapture-utils.test.ts

```diff
@@ -158,6 +158,108 @@ describe(`Autocapture utility functions`, () => {
+                    },
+                    true,
+                ],
+                [
+                    'when the click target is matching in the allow list',
+                    makeSingleBranchOfDomTree([{ tag: 'div' }, { tag: 'button' }, { tag: 'svg', id: 'in-allowlist' }]),
+                    {
+                        css_selector_allowlist: ['[id]'],
+                    },
+                    true,
+                ],
+                [
+                    'when the parent does not match the allowlist',
+                    makeSingleBranchOfDomTree([
+                        { tag: 'div' },
+                        { tag: 'button', id: '[id=not-the-configured-value]' },
+                        { tag: 'svg' },
+                    ]),
+                    {
+                        // the click was detected on the SVG, but the button is not in the allow list,
+                        // so we should detect the click
+                        css_selector_allowlist: ['in-allowlist'],
+                    },
+                    false,
+                ],
+                [
+                    'when the click target (and none of its parents) does not match the allowlist',
+                    makeSingleBranchOfDomTree([
+                        { tag: 'div' },
+                        { tag: 'button' },
+                        { tag: 'svg', id: '[id=not-the-configured-value]' },
+                    ]),
+                    {
+                        css_selector_allowlist: ['in-allowlist'],
+                    },
+                    false,
+                ],
+                [
+                    'when combining allow lists',
+                    makeSingleBranchOfDomTree([{ tag: 'div' }, { tag: 'button', id: 'in-allowlist' }, { tag: 'svg' }]),
+                    {
+                        // the tree for the click does have an id
+                        css_selector_allowlist: ['[id]'],
+                        // but we only detect if there is an img in the tree
+                        element_allowlist: ['img'],
+                    },
+                    false,
+                ],
+                [
+                    'combine allow lists - but showing it considers them separately',
+                    makeSingleBranchOfDomTree([
+                        { tag: 'div' },
+                        { tag: 'button', id: 'in-allowlist' },
+                        { tag: 'img' },
+                        { tag: 'svg' },
+                    ]),
+                    {
+                        // the tree for the click does have an id
+                        css_selector_allowlist: ['[id]'],
+                        // and the tree for the click does have an img
+                        element_allowlist: ['img'],
+                    },
+                    true,
+                ],
+            ])('correctly respects the allow list: %s', (_, clickTarget, autoCaptureConfig, shouldCapture) => {
```

Comment:
i parameterised the tests in a single commit in case folk don't agree that it's more readable this way